src/Abstractions/Entities/TaskEntityContext.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.DurableTask.Entities;
+
+/// <summary>
+/// The context for a <see cref="ITaskEntity"/>.
+/// </summary>
+public abstract class TaskEntityContext
+{
+    /// <summary>
+    /// Gets the instance ID of this entity.
+    /// </summary>
+    public abstract EntityInstanceId Id { get; }
+
+    /// <summary>
+    /// Signals an entity operation.
+    /// </summary>
+    /// <param name="id">The entity to signal.</param>
+    /// <param name="operationName">The operation name.</param>
+    /// <param name="input">The operation input.</param>
+    /// <param name="options">The options to signal with.</param>
```

Comment:
would it make sense to mention here that it is possible to pass a DateTimeOffset directly (thanks to the implicit conversion)?

Reply:
I would prefer to leave that for discovery by looking at the SignalEntityOptions type itself. Purely because if I add it here, I should be adding it to all the other usages of it for consistency. And that seems like a lot of extra comments for little value.